### ISSUE TITLE
IS2_INCONSISTENT_SYNC false-positive

### ISSUE THREAD

**Issue by boris-petrov** [spotbugs/spotbugs#1051]
```java
public class Foo  {
	private int a = 0;

	public synchronized void update(int a) {
		this.a = a;
	}

	public synchronized void foo(List<Integer> list) {
		if (list.stream().allMatch(item -> item < a)) {
			return;
		}
	}
}
```

This code gives a `IS2_INCONSISTENT_SYNC` warning for the `a` variable which is wrong - the method is synchronized and this access is safe.

**Comment by ThrawnCA** [comment]
I'm not so sure; can someone weigh in with more experience about how lambdas are turned into bytecode?

**Comment by iloveeclipse** [comment]
Lambda call itself is not synchronized, this is why spotbugs flags it as warning, but it is a false positive because lambda is called from same thread inside the synchronized block. A way to "mute" this would be to extract comparison into a synchronized method, but I would just add a filter here.

**Comment by boris-petrov** [comment]
Yes, of course lambdas are generally not synchronized but Java streams are ubiquitous (and synchronous unless parallel) so an exception for them would be nice. If you think it's not worth doing in SpotBugs, feel free to close the issue. Thanks!

**Comment by WddVictor** [comment]
Maven SpotBugs Plugin: 3.1.12.2 cannot reproduce this.

**Comment by boris-petrov** [comment]
@WddVictor

```gradle
plugins {
	id 'java'
	id 'com.github.spotbugs' version '3.0.0'
}

sourceCompatibility = JavaVersion.VERSION_13
targetCompatibility = JavaVersion.VERSION_13

repositories {
	mavenCentral()
	jcenter()
}

dependencies {
	spotbugs 'com.github.spotbugs:spotbugs:4.0.0-beta4'
}

spotbugs {
	effort = 'max'
	reportLevel = 'low'
}
```

This gives the warning on the example on the first post.
